### ISSUE TITLE
/ver reporting error about "java.lang.NumberFormatException: For input string: "<html>"

### ISSUE THREAD

**Issue by malonnnn** [PaperMC/Paper#1080]
Every time I do /version in my server, it tries to check github for any updates, but can never seemingly check correctly.

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.4 LTS
Release:	16.04
Codename:	xenial
```

```
$ uname -a
Linux localhost 4.15.12-x86_64-linode105 #1 SMP Thu Mar 22 02:13:40 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

```
$ java -version
java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)
```

```
ver
[10:27:13 INFO]: This server is running Paper version git-Paper-1389 (MC: 1.12.2) (Implementing API version 1.12.2-R0.1-SNAPSHOT)
[10:27:13 INFO]: Checking version, please wait...
[10:27:13 WARN]: java.lang.NumberFormatException: For input string: "<html>"
[10:27:13 WARN]:        at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
[10:27:13 WARN]:        at java.lang.Integer.parseInt(Integer.java:580)
[10:27:13 WARN]:        at java.lang.Integer.valueOf(Integer.java:740)
[10:27:13 WARN]:        at java.lang.Integer.decode(Integer.java:1197)
[10:27:13 WARN]:        at org.bukkit.command.defaults.VersionCommand.getFromJenkins(VersionCommand.java:277)
[10:27:13 WARN]:        at org.bukkit.command.defaults.VersionCommand.getDistance(VersionCommand.java:248)
[10:27:13 WARN]:        at org.bukkit.command.defaults.VersionCommand.obtainVersion(VersionCommand.java:196)
[10:27:13 WARN]:        at org.bukkit.command.defaults.VersionCommand.access$000(VersionCommand.java:36)
[10:27:13 WARN]:        at org.bukkit.command.defaults.VersionCommand$1.run(VersionCommand.java:181)
[10:27:13 WARN]:        at java.lang.Thread.run(Thread.java:748)
[10:27:13 INFO]: Unknown version
```

**Comment by zachbr** [comment]
Is this something you're continuing to see or just something you saw once?

**Comment by malonnnn** [comment]
I've been seeing it constantly, it's reproducible every time I run /ver (only when the response isn't cached, if it's cached then it just immediately says "Unknown Version" without the console error) and it's been happening for the last 10 to 15 builds. It started out of the blue one day, and I've done everything I can think of on my end, I even did a fresh install configuring a new VPS in a different datacenter, same issue. It's been driving me nuts, I just assumed it was something on my end, but I've ruled out just about everything on my end that I can find. Suggestions are welcome.

**Comment by zachbr** [comment]
I've linked a modified paperclip.jar that I want you to run. It includes two changes:

1. It uses the updated version check URL without redirects - just in case there's something about your system/network configuration that is causing issues with redirects.
2. If it still fails, it will write the contents of the page it is getting to the logger. This will make a mess but it will at least be able to tell us if its jenkins screwing up, cloudflare screwing up, or something else.

https://keybase.pub/zachbr/paper/github/issues/1080/paperclip.jar

Please let us know if this works for you. If it doesn't please attach the log output.

**Comment by malonnnn** [comment]
This version works fine and properly reports the version

Does this mean it's on my end? If so, how can I check what's going on and what do I look for/where do I begin looking?

**Comment by zachbr** [comment]
I really don't know. We made these redirect switches a while ago and you are the first person to report any problems with them. I couldn't tell you if this is directly on your end or on your providers end, their peers, end, somewhere buried on my end (though I looked and can't see anything blatant).

I am going to push a commit to master in a few minutes that will make this change for everyone and then we can just not worry about it anymore.
